### PR TITLE
[Concept] New harvesting capabilities

### LINKED ISSUE #80: [Concept] New harvesting capabilities
Document https://github.com/nemtech/catapult-server/issues/15

## Deliverables

* Adapt [Harvesting](https://nemtech.github.io/concepts/harvesting.html) concept
* Rename [beneficiaryPublicKey] in the block header (https://nemtech.github.io/concepts/block.html) 
* Update the swagger.json file.

## Dependencies

-  https://github.com/nemtech/catapult-server/issues/15

### PR DESCRIPTION
**Pull request by dgarcia360**
Closes #80

### CODE REVIEW

**Comment by dgarcia360** [comment]
Closes #80. Merge when the feature is available on catapult-server.

File: source/concepts/harvesting.rst [hunk 1]
```diff
@@ -3,55 +3,73 @@ Harvesting
...
     :delim: ;
 
-    **Configuration** ; Setup catapult-server node.; Activate remote harvesting.
-    **Cost** ; The node maintenance (electricity, cost VPN).; The transaction fee.
-    **Security**; The private key is stored in the node.; A proxy private key is shared with node.
-    **Reward**; Equal.; Equal.
+    **Configuration** ; Setup node.; Activate remote harvesting.
+    **Cost** ; The node maintenance (electricity, cost VPN).; The activation transaction fee.
+    **Security**; The node stores the private key.;  A proxy private key is shared with a node.
+    **Reward**; Complete. The node owner can share part of the reward with a beneficiary account.; Total reward - beneficiary share.
```
Comment (ivyfung1) [inline source/concepts/harvesting.rst]: Complete -> Total reward

File: source/concepts/harvesting.rst [hunk 2]
```diff
@@ -3,55 +3,73 @@ Harvesting
...
+The harvesting :doc:`account <account>` - called the harvester - gets the fees for the :doc:`transactions <transaction>` in the block. This reward gives the harvester an incentive to add as many transactions to the block as possible.
 
-Catapult software allows defining any :doc:`mosaic <mosaic>` for harvesting, using its configurable
-properties to fit the business needs. For example, consortium networks can distribute harvesting mosaics between the
-companies that are running the infrastructure, while other participants need to pay fees in the currency mosaic to consume services.
+********************
+Eligibility criteria
+********************
+
+The :doc:`account importance <account>` determines the chances to create a new block. The importance is calculated as the relation between the number of :ref:`harvesting mosaics <harvesting-mosaic>` the account owns and the total supply available.
```
[the remark this answers is not in the document]
Comment (dgarcia360) [reply source/concepts/harvesting.rst]: I have changed it, let me know what do you think.

**Comment by dgarcia360** [comment]
Thanks @ivyfung1 for your revision!

**Pull request merged**